sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1666,6 +1684,82 @@ def get_custom_materialization_type(name: str) -> t.Type[CustomMaterialization]:
     return strategy_type
 
 
+class EngineManagedStrategy(MaterializableStrategy):
+    def create(
+        self,
+        table_name: str,
+        model: Model,
+        is_table_deployable: bool,
+        is_snapshot_deployable: bool,
+        **render_kwargs: t.Any,
+    ) -> None:
+        if is_table_deployable and is_snapshot_deployable:
+            # We could deploy this to prod; create a proper managed table
+            logger.info("Creating managed table: %s", table_name)
+            self.adapter.create_managed_table(
+                table_name=table_name,
+                query=model.render_query_or_raise(**render_kwargs),
+                columns_to_types=model.columns_to_types,
+                partitioned_by=model.partitioned_by,
+                clustered_by=model.clustered_by,
+                table_properties=model.physical_properties,
+                table_description=model.description,
+                column_descriptions=model.column_descriptions,
+            )
+        else:
+            # Otherwise, fallback to the default logic of creating normal tables
+            super().create(
+                table_name=table_name,
+                model=model,
+                is_table_deployable=is_table_deployable,
+                is_snapshot_deployable=is_snapshot_deployable,
+                **render_kwargs,
+            )
+
+    def insert(
+        self,
+        table_name: str,
+        query_or_df: QueryOrDF,
+        model: Model,
+        is_first_insert: bool,
+        **kwargs: t.Any,
+    ) -> None:
+        deployability_index: DeployabilityIndex = kwargs["deployability_index"]
+        snapshot: Snapshot = kwargs["snapshot"]
+        is_snapshot_deployable = deployability_index.is_deployable(snapshot)
+
+        if is_first_insert:
```

Comment:
```
if is_first_insert and not is_snapshot_deployable:
```

Reply:
Btw, why do we do this only on the first insert? Shouldn't we always update the dev table in dev?

Reply:
My lack of knowledge around exactly when and how `insert()` is called multiple times.

Managed models are not incremental. The main model would only be created once if it was deployable, so I figured we should only populate the dev preview table once to match?

If the user changes something, isnt that a new snapshot / new table?

Reply:
I guess there is no harm is doing it for all `insert()`'s, i've adjusted the logic